### PR TITLE
[WIP] Changing the Bot to Event-Based

### PR DESCRIPTION
**Pull request by jtdroste**
# NOTE: Still WIP
### Changes:
- Create EventManager / IEventHandler
- Moved out a bunch of the cell things into handlers
### TODO
- [ ] Add autoloading of the pokemongo_bot/handlers/*.py directory
- [ ] Setup the stepper to use events
- [ ] Possibly add bot start / bot shutdown events

@OpenPoGo/maintainers

### CODE REVIEW

File: pokemongo_bot/handlers/catch_handler.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+from cell_workers import PokemonCatchWorker
+from cell_workers.utils import distance
+from events import IEventHandler
+
+class CachablePokemonHandler(IEventHandler):
```
Comment (douglascamata) [inline pokemongo_bot/handlers/catch_handler.py]: typo here

**Comment by MaxLeiter** [comment]
Mind providing a list of what's left to be worked on so other people can contribute? Awesome work so far :)

**Comment by TheSavior** [comment]
I strongly disagree with this PR

**Comment by jtdroste** [comment]
@TheSavior can you join our [slack](https://openpogo-slack.herokuapp.com/) and maybe discuss a bit about this?

File: pokemongo_bot/handlers/catch_handler.py (hunk 1 above)
Comment (jtdroste) [inline pokemongo_bot/handlers/catch_handler.py]: ty - I was thinking of cache.  Note, this is still heavily WIP - I just wanted to add some progress

**Comment by elicwhite** [comment]
Sorry, didn't realize this was a different bot. I have no feelings about the approach you guys take, but I strongly don't think this is the right approach for the upstream bot. I have laid out the concerns that I have about it here: https://github.com/PokemonGoF/PokemonGo-Bot/pull/706

The approach we want to go in to enable extensibility and configuration is using behavior trees https://github.com/PokemonGoF/PokemonGo-Bot/issues/142

**Comment by jtdroste** [comment]
#67 - closed...

**Pull request closed without merging**